pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -20,7 +20,10 @@
 # TODO enable shared libs for cross-compiling
 , enableSharedExecutables ? !isCross && (((ghc.isGhcjs or false) || stdenv.lib.versionOlder "7.7" ghc.version))
 , enableSharedLibraries ? !isCross && (((ghc.isGhcjs or false) || stdenv.lib.versionOlder "7.7" ghc.version))
-, enableSplitObjs ? !stdenv.isDarwin # http://hackage.haskell.org/trac/ghc/ticket/4013
+# For GHC 8.0.2 we use split-sections instead
+# Unsupported for Darwin/Clang, see http://hackage.haskell.org/trac/ghc/ticket/4013
+, enableSplitObjs ? (!stdenv.isDarwin && (stdenv.lib.versionAtLeast "8.0.1" ghc.version))
+, enableSplitSections ? (!stdenv.isDarwin && (stdenv.lib.versionOlder "8.0.1" ghc.version))
```

Comment:
Yes, that sounds like a good compromise. We can also abstract the issue somewhat by adding an appropriate `haskell.lib.dontDeadStrip` function that hides the implementation. If it turns out that we need to control those flags in dozens of packages, then I'm all for (a) adding an appropriate option to the generic builder and (b) teaching cabal2nix to generate them when necessary.